### ISSUE TITLE
exl2 of merge loads in with 2K truncation length instead of 8K for GGUF?

### ISSUE THREAD

**Issue by jim-plus** [oobabooga/text-generation-webui#5750]
### Describe the bug

I've not uploaded the exl2 quant yet because of the issue, but the safetensors version is here:
https://huggingface.co/grimjim/kunoichi-lemon-royale-7B

In short, I am getting a truncation length of 2048 instead of 8192 when loading the 8.0bpw h8 exl2 in ooba, but get the full 8192 when loading the Q8_0 GGUF equivalent. Updating ooba did not help, though I wonder if the exllamav2 components need a refresh to sync with upstream.

Below is the config.json of the loaded exl2; I manually corrected the calibration length from 2048 to 8192, but that did not resolve the issue. There is no apparent origin for the 2048 truncation value upon loading. The Q8_0 GGUF of the model correctly loads with 8192 context length.
```
{
    "_name_or_path": "SanjiWatsuki/Kunoichi-7B",
    "architectures": [
        "MistralForCausalLM"
    ],
    "attention_dropout": 0.0,
    "bos_token_id": 1,
    "eos_token_id": 2,
    "hidden_act": "silu",
    "hidden_size": 4096,
    "initializer_range": 0.02,
    "intermediate_size": 14336,
    "max_position_embeddings": 8192,
    "model_type": "mistral",
    "num_attention_heads": 32,
    "num_hidden_layers": 32,
    "num_key_value_heads": 8,
    "rms_norm_eps": 1e-05,
    "rope_theta": 10000.0,
    "sliding_window": 4096,
    "tie_word_embeddings": false,
    "torch_dtype": "bfloat16",
    "transformers_version": "4.38.2",
    "use_cache": true,
    "vocab_size": 32000,
    "quantization_config": {
        "quant_method": "exl2",
        "version": "0.0.16",
        "bits": 8.0,
        "head_bits": 6,
        "calibration": {
            "rows": 100,
            "length": 8192,
            "dataset": "(default)"
        }
    }
}
```

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Currently: download safetensors version, quant with exllamav2 to get 8.0bpw h8, then load in ooba.

### Logs

```shell
12:15:34-126265 INFO     Loading "kunoichi-lemon-royale-7B-8.0bpw_h8_exl2"
12:15:47-116039 INFO     LOADER: "ExLlamav2_HF"
12:15:47-117824 INFO     TRUNCATION LENGTH: 2048
12:15:47-118828 INFO     INSTRUCTION TEMPLATE: "Alpaca"
12:15:47-119831 INFO     Loaded the model in 12.99 seconds.
```

### System Info

```shell
Windows 11, Nvidia RTX4060Ti 16GB.
```

**Comment by turboderp** [comment]
I've identified what I think is the issue. The problem is the `quantization_config` key added in more recent EXL2 conversion which lists the number of bits used to convert the model. In EXL2 this value is a float, but TGW parses it as an int leading to an exception in `update_model_parameters`. The function is then aborted before it reaches the `max_seq_len` key and so it never updates that and uses the default value of 2048 instead.

A quick fix would be to change line 199 of modules/models_settings.py from:

```python
            value = int(value)
```

to

```python
            value = int(float(value))
```

This seems to work here, but I don't have a clear picture of where else the `wbits` entry could be used, or if a better way would be to not interpret "quantization_config.bits" -> "wbits" for EXL2 models when reading the config. @oobabooga ?

**Comment by Danweel** [comment]
Hi, seems I ran into this issue with a newer version of the code: 

`def update_model_parameters(state, initial=False):
    '''
    UI: update the command-line arguments based on the interface values
    '''
    elements = ui.list_model_elements()  # the names of the parameters

    for i, element in enumerate(elements):
        if element not in state:
            continue

        value = state[element]
        if initial and element in shared.provided_arguments:
            continue

        if element == 'cpu_memory' and value == 0:
            value = vars(shared.args_defaults)[element]

        setattr(shared.args, element, value)`

Don't want to just blindly change value from the above to `value = int(float(value))`, now that it's no longer looking like it's set to 'value = int(value)'. My limited understanding makes me think this `vars(shared.args_defaults)` now points to a yaml file in the path?

Looking at the config-user.yaml (which it looks like shared.args might be pointing to), I notice on line 1 it hasn't updated to the model I'm using now - it still lists the very first one I loaded. So far, I've only ever used the UI - never the prompt. Is this an example of the UI not updating something it should? What could be causing that? How do I manually fix it?

Thanks for any insight, and if it might help someone else.
